core/src/main/scala/org/http4s/headers/`Access-Control-Allow-Methods`.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.headers
+
+import cats.data.NonEmptyList
+import org.http4s.internal.parsing.Rfc7230
+import org.http4s.{Header, Method, ParseResult}
+import org.typelevel.ci.{CIString, CIStringSyntax}
+
+/** The `Access-Control-Allow-Methods` header. */
+sealed trait `Access-Control-Allow-Methods`
+
+object `Access-Control-Allow-Methods` {
+
+  private val Wildcard: CIString = CIString("*")
+
+  /** The value "*" only counts as a special wildcard value for requests without credentials
+    * (requests without HTTP cookies or HTTP authentication information). In requests with
+    * credentials, it is treated as the literal method name "*" without special semantics.
+    */
```

Comment:
The spec is in the [fetch living standard](https://fetch.spec.whatwg.org/#http-access-control-allow-methods) rather than an RFC - the relevant extract is [here](https://fetch.spec.whatwg.org/#http-new-header-syntax):

> For `Access-Control-Expose-Headers`, `Access-Control-Allow-Methods`, and `Access-Control-Allow-Headers` response headers, the value `*` counts as a wildcard for requests without credentials. For such requests there is no way to solely match a header name or method that is `*`.

It's not as explicit as it could be about the non-wildcard case, so there's probably no harm in quoting the MDN doc in addition.